atf_core/src/atf_core/recorder.py

```diff
@@ -60,24 +61,34 @@ def __init__(self, test):
         rospy.on_shutdown(self.shutdown)
         
         # wait for topics, services, actions and tfs to become active
+        wait_timeout = None # None is infinite timeout
+        start_time = rospy.Time.now()
+        if test.robot_config != None and 'wait_timeout' in test.robot_config:
+            if test.robot_config["wait_timeout"] > 0:
+                wait_timeout = test.robot_config["wait_timeout"]
+                rospy.loginfo("wait_timeout is set to %s", str(wait_timeout))
+
         if test.robot_config != None and 'wait_for_topics' in test.robot_config:
             for topic in test.robot_config["wait_for_topics"]:
+                self.check_for_timeout(start_time, wait_timeout)
                 rospy.loginfo("Waiting for topic '%s'...", topic)
-                rospy.wait_for_message(topic, rospy.AnyMsg)
+                rospy.wait_for_message(topic, rospy.AnyMsg, wait_timeout)
```

Comment:
when adding the timeout here, the above call to `check_for_timeout` is obsolete

Reply:
done in https://github.com/floweisshardt/atf/pull/186

Reply:
It's not the same if you have to wait for 10 topics. I considder the `wait_timeout` to be the maximum overall time we want to wait for the test to start, not the maximum time per topic. E.g. if you have to wait 60sec for 10 topics you end up in 600sec before the test aborts.

Same for a combination of waiting for services, action, topics, tf and diagnostics. If the `wait_timeout` is set to 60s I expect the test to fail after (or at least close to) 60s, not depending on how many topics/services/tfs... are configured.

Reply:
lets discuss the remaining issues and my contributions from https://github.com/floweisshardt/atf/pull/186 in person....